### DISCUSSION TITLE
Add libraries to the torch environment

### DISCUSSION THREAD

**Discussion by shaido987** [PaddlePaddle/PaddleSpatial#78] [BaiduKDDCup2022] [answered]
Hello,

is it possible to add the following two libraries to the torch environment?

- pytorch-geometric-temporal
- darts (installed with u8darts[torch] to access torch models)

**Comment by ramber1836** [comment]
Yes, we will work on it shortly, and the online test will be updated in a
couple of days due to the hard workload.
Thanks for your understanding.

Regards,
Xinjiang Lu

**Reply by shaido987** [reply]
Sure, thanks for the help. Could you update here when the libraries are added?

**Reply by ramber1836** [reply]
Yes, I will notify you here once it is ready.

**Reply by ramber1836** [reply]
@shaido987 These dependencies have been added to the pytorch environment in the online test.

**Reply by shaido987** [reply]
@ramber1836 Thanks for the help! I'm still getting an submit error when trying ('NoneType' object has no attribute 'forecast'), could you check what is going wrong? Team name: didadida_hualahuala

**Reply by shaido987** [reply]
@ramber1836 I tried with a very simple predict file that simply returns 1s with the correct dimensions. This works. But if I add an import for torch_geometric_temporal in the predict file then it fails with an error: "异常". It could be some issue with the dependency installation? (darts seem to be working correctly.)

**Comment by ramber1836** [comment]
You can try it again, the backend docker image has been updated.

Regards,
Xinjiang Lu

**Reply by shaido987** [reply]
@ramber1836 Hello, even with the updated docker image I get the error "异常" when simply importing torch_geometric_temporal without using it. The same predict file without the import works correctly. Could you check what is wrong? (Team name: didadida_hualahuala, you can see the submission at 2022-05-25 09:35)

**Reply by ramber1836** [reply]
The real error is that "std::bad_alloc" when importing torch_geometric_temporal library. 
What is the memory setup in your local machine?

**Reply by shaido987** [reply]
@ramber1836 It should be an issue with the versions of the library dependencies, they need to match the installed pytorch and cuda versions: https://stackoverflow.com/questions/72039582/terminate-called-after-throwing-an-instance-of-stdbad-alloc-from-importing-t

**Reply by shaido987** [reply]
For the installed pytorch I believe it should be:
```
pip install torch-scatter -f https://data.pyg.org/whl/torch-1.9.0+cu102.html
pip install torch-sparse -f https://data.pyg.org/whl/torch-1.9.0+cu102.html
pip install torch-geometric
pip install torch-geometric-temporal
```

**Comment by ramber1836** [comment, marked as answer]
The pytorch and CUDA in the online test environment are 1.11 and 11.2,
respectively.
Is torch-geometric-temporal compatible with CUDA 11.2 ?

**Reply by shaido987** [reply]
From what I have been reading, it should work with the 11.3 version as the major version match. So:
```
pip install torch-scatter -f https://data.pyg.org/whl/torch-1.11.0+cu113.html
pip install torch-sparse -f https://data.pyg.org/whl/torch-1.11.0+cu113.html
pip install torch-geometric
pip install torch-geometric-temporal
```

Btw, the pytorch version specified at https://github.com/PaddlePaddle/PaddleSpatial/blob/main/apps/wpf_baseline_gru/kddcup22-sdwpf-evaluation/preinstalled-pkgs/pytorch_env_installed_packages.md seems to be incorrect.

**Reply by ramber1836** [reply]
Seems not working 
<img width="1122" alt="5e39a6f0c43a7db2f4e92e2c6" src="https://user-images.githubusercontent.com/19561583/170295054-a95989d8-c1d7-4eb1-a140-c2ee9417dd54.png">

**Reply by shaido987** [reply]
That's too bad. Thanks for the great help anyway. I will try to see if I can solve it in some other way.
